include/dlaf/permutations/general/perms.h

```diff
@@ -27,16 +27,17 @@ template <class T, Coord coord>
 void applyPermutationsOnDevice(
     GlobalElementIndex out_begin, GlobalElementSize sz, SizeType in_offset,
     const matrix::Distribution& distr, const SizeType* perms,
-    const std::vector<pika::shared_future<matrix::Tile<const T, Device::GPU>>>& in_tiles,
+    const std::vector<matrix::internal::TileAsyncRwMutexReadOnlyWrapper<T, Device::GPU>>& in_tiles,
```

Comment:
Because a `ReadOnlyTileSender` does not send a `ReadOnlyTileSender`.

This was one of the good and bad things of `shared_future<Tile<...>>` (with `keep_future`). With a `shared_future`
- The sender sends itself and you don't have to deal with two different types.
- The sender sends itself and you don't know if the sender you're dealing with is a ready sender or not just by looking at the type (you only know at runtime or from its use).

`applyPermutationsOnDevice` now takes a `TileAsyncRwMutexReadOnlyWrapper<...>`. That is the type sent by a `ReadOnlyTileSender` and is not a sender in itself.

C.f. https://github.com/msimberg/DLA-Future/blob/2bed2f2d7081f23c6308c251f010a73134999f22/include/dlaf/matrix/tile.h#L161.